src/Fetcher/TululaFetcher.php

```diff
@@ -0,0 +1,241 @@
+<?php
+
+/*
+ * This file is part of the Starfleet Project.
+ *
+ * (c) Starfleet <msantostefano@jolicode.com>
+ *
+ * For the full copyright and license information,
+ * please view the LICENSE file that was distributed with this source code.
+ */
+
+namespace App\Fetcher;
+
+use App\Entity\Conference;
+use App\Entity\Continent;
+use App\Entity\ExcludedTag;
+use App\Entity\Tag;
+use App\Repository\ExcludedTagRepository;
+use App\Repository\TagRepository;
+use Psr\Log\LoggerInterface;
+use Symfony\Component\String\Slugger\SluggerInterface;
+use Symfony\Contracts\HttpClient\Exception\TransportExceptionInterface;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+
+class TululaFetcher implements FetcherInterface
+{
+    const SOURCE = 'tulula';
+    private const TAGS_SYNONYMS = [
+        'android',
+        'css',
+        'data',
+        'devops',
+        'dotnet',
+        'facebook',
+        'golang',
+        'html',
+        'ios',
+        'java',
+        'javascript',
+        'nodejs',
+        'php',
+        'python',
+        'react native',
+        'ruby',
+        'rust',
+        'security',
+        'ux',
+        'ai/ml',
+        'sql',
+        'iot',
+    ];
+
+    private LoggerInterface $logger;
+    private TagRepository $tagRepository;
+    /** @var ExcludedTag[] */
+    private array $excludedTags;
+    private LocationGuesser $locationGuesser;
+    private SluggerInterface $slugger;
+    private HttpClientInterface $client;
+
+    public function __construct(LoggerInterface $logger, LocationGuesser $locationGuesser, HttpClientInterface $client, TagRepository $tagRepository, ExcludedTagRepository $excludedTagRepository, SluggerInterface $slugger)
+    {
+        $this->client = $client;
+        $this->logger = $logger;
+        $this->tagRepository = $tagRepository;
+        $this->locationGuesser = $locationGuesser;
+        $this->slugger = $slugger;
+        $this->excludedTags = $excludedTagRepository->findAll();
+    }
+
+    public function isActive(): bool
+    {
+        return true;
+    }
+
+    public function getUrl(array $params = []): string
+    {
+        return 'https://tulu.la/api/public';
+    }
+
+    public function fetch(): \Generator
+    {
+        foreach ($this->tagRepository->findBy(['selected' => true]) as $tag) {
+            $tagKey = array_search($this->slugger->slug(strtolower($tag->getName()))->toString(), self::TAGS_SYNONYMS);
+
+            if (false === $tagKey) {
+                continue;
+            }
+
+            $tagSynonym = self::TAGS_SYNONYMS[$tagKey];
```

Comment:
I don't understand this code at all. it's basically the same thing than an \in_array ?